packages/dev/serializers/src/glTF/2.0/glTFMaterialExporter.ts

```diff
@@ -84,240 +64,89 @@ function getFileExtensionFromMimeType(mimeType: ImageMimeType): string {
-        // Defines a cubic bezier curve where x is specular power and y is roughness
-        const P0 = new Vector2(0, 1);
-        const P1 = new Vector2(0, 0.1);
-        const P2 = new Vector2(0, 0.1);
-        const P3 = new Vector2(1300, 0.1);
-
-        /**
-         * Given the control points, solve for x based on a given t for a cubic bezier curve
-         * @param t a value between 0 and 1
-         * @param p0 first control point
-         * @param p1 second control point
-         * @param p2 third control point
-         * @param p3 fourth control point
-         * @returns number result of cubic bezier curve at the specified t
-         */
-        function cubicBezierCurve(t: number, p0: number, p1: number, p2: number, p3: number): number {
-            return (1 - t) * (1 - t) * (1 - t) * p0 + 3 * (1 - t) * (1 - t) * t * p1 + 3 * (1 - t) * t * t * p2 + t * t * t * p3;
-        }
+    return rawTexture;
+}
 
-        /**
-         * Evaluates a specified specular power value to determine the appropriate roughness value,
-         * based on a pre-defined cubic bezier curve with specular on the abscissa axis (x-axis)
-         * and roughness on the ordinant axis (y-axis)
-         * @param specularPower specular power of standard material
-         * @returns Number representing the roughness value
-         */
-        function solveForRoughness(specularPower: number): number {
-            // Given P0.x = 0, P1.x = 0, P2.x = 0
-            //   x = t * t * t * P3.x
-            //   t = (x / P3.x)^(1/3)
-            const t = Math.pow(specularPower / P3.x, 0.333333);
-            return cubicBezierCurve(t, P0.y, P1.y, P2.y, P3.y);
+function ConvertPixelArrayToFloat32(pixels: ArrayBufferView): Float32Array {
+    if (pixels instanceof Uint8Array) {
+        const length = pixels.length;
+        const buffer = new Float32Array(pixels.length);
+        for (let i = 0; i < length; ++i) {
+            buffer[i] = pixels[i] / 255;
         }
+        return buffer;
+    } else if (pixels instanceof Float32Array) {
+        return pixels;
+    } else {
+        throw new Error("Unsupported pixel format!");
+    }
+}
 
-        const diffuse = babylonStandardMaterial.diffuseColor.toLinearSpace(babylonStandardMaterial.getScene().getEngine().useExactSrgbConversions).scale(0.5);
-        const opacity = babylonStandardMaterial.alpha;
-        const specularPower = Scalar.Clamp(babylonStandardMaterial.specularPower, 0, _GLTFMaterialExporter._MaxSpecularPower);
-
-        const roughness = solveForRoughness(specularPower);
-
-        const glTFPbrMetallicRoughness: IMaterialPbrMetallicRoughness = {
-            baseColorFactor: [diffuse.r, diffuse.g, diffuse.b, opacity],
-            metallicFactor: 0,
-            roughnessFactor: roughness,
-        };
+/**
+ * Utility methods for working with glTF material conversion properties.
+ * @internal
+ */
+export class GLTFMaterialExporter {
+    private _exporter: GLTFExporter;

```

Comment:
Seems like all these properties should be readonly?
```suggestion
    private readonly _exporter: GLTFExporter;
```